### ISSUE TITLE
Differentiable quantum layer (parameter-shift autograd for PyTorch)

### ISSUE THREAD

**Issue by ddri** [marqov-dev/marqov-sdk#53]
Part of epic **marqov-dev/marqov-platform#1340** (Extract value from NVIDIA CUDA-Q tutorial).

## Problem

The SDK is circuit-submission focused (define circuit → execute → counts). It has **no differentiable-quantum-layer story** — you can't drop a quantum expectation value into a PyTorch model and train it end-to-end. That rules out hybrid quantum-classical ML workflows, a growing use case.

## Reference implementation

`II_LRZ-HPCQC/example2/hybrid_qnn.py` implements exactly this pattern with a custom `torch.autograd.Function`:

- **`QuantumExpectation(torch.autograd.Function)`** (`hybrid_qnn.py:188-226`) — forward runs the circuit for expectation values; backward implements the **parameter-shift rule**:
  ```python
  t_plus  = theta.clone(); t_plus[:, i]  += shift
  t_minus = theta.clone(); t_minus[:, i] -= shift
  derivative = 0.5 * (qc.run(t_plus) - qc.run(t_minus))
  ```
- **`QuantumCircuitRunner`** (`hybrid_qnn.py:115-186`) — wraps the kernel + observable; shot-based expectation (realistic noise) with sync `run()` and async `run_async()`/`collect_async()`.
- **Hybrid model** (`hybrid_qnn.py:229-243`) — classical encoder (MLP) → quantum layer → classical head; trains on CPU/GPU, can infer on QPU.

Validated: trains 10 epochs, MSE 2.04 → 0.47.

## Proposal

Add an optional `marqov.ml` (or `marqov.torch`) module providing a backend-agnostic differentiable quantum layer:
- A `torch.autograd.Function` computing expectation values via any `marqov` executor, with parameter-shift gradients.
- Works over the executor abstraction so gradients can be evaluated on **any** backend (local sim, GPU sim, QPU), not just CUDA-Q.

## Acceptance criteria

- [ ] `QuantumExpectation`-style autograd Function over `BaseExecutor`
- [ ] Parameter-shift gradient (configurable shift, default π/2)
- [ ] Batched forward via async executor dispatch (see marqov-dev/marqov-sdk#54)
- [ ] Example: fit a 1-D function with a hybrid model (port of the tutorial's sine-fit)
- [ ] `marqov[torch]` optional extra

## Notes

- Deliberately framework-agnostic: the value is a reusable "bridge external/non-autodiff compute into an autograd graph" pattern, not a CUDA-Q-specific one.
- Lower priority than the executor/capsule items; ship after marqov-dev/marqov-sdk#52 if hybrid-ML is on the roadmap.
